Creación de las interfaces de los DTO

user-ijavieh opened the pull request:
Creación de las interfaces de los Data Tranfer Objects según la documentación de la api.

Se crearon otros objetos como OrderBaseDTO o AllergenBaseDTO que se utilizan para evitar ciclos infinitos.

Además se creo la carpeta src/app/interfaces donde se encuentran todos los nuevos cambios

user-ijavieh committed:
Creación de las interfaces de los DTO

Creación de las interfaces de los Data Tranfer Objects según la documentación de la api.

FranWDev commented on frontend/src/app/interfaces/Allergen/Allergen.ts:
Se importa una receta base pero la interfaz del archivo no es heredada de ella

FranWDev commented on frontend/src/app/interfaces/Order/Order.ts:
```diff
@@ -0,0 +1,11 @@
+
+export interface Order {
+  id: number;
+  user: UserResponseDTO;
```
Se debe usar la id del usuario unicamente, citado del backend:
```
@ManyToOne
@JoinColumn(name = "id_usuario") <- se relaciona por la ID, el usuario es completado por Hibernate
private User user;
```
```
public class OrderRequestDTO {

    @NotNull(message = "El ID del usuario no puede ser nulo")
    @Schema(description = "Identificador único del usuario asociado al pedido", example = "4", required = true)
    private Integer userId; <- Aqui se ve la ID en la request
```

FranWDev commented on frontend/src/app/interfaces/Order/Order.ts:
Incongruencias con la interfaz padre, se debe extender y evitar datos duplicados como en Allergen

FranWDev commented on frontend/src/app/interfaces/Order/OrderBaseDTO.ts:
```diff
@@ -0,0 +1,7 @@
+  id: number;
+  userId: number;
+  userName: string;
+  orderDate: string;
```
OrderDate no es una String, es un objeto LocalDateTime, que no existe en typescript, importante anotar este formato:
`YYYY-MM-DDTHH:mm:ss (ISO 8601) `

FranWDev commented on frontend/src/app/interfaces/Product.ts:
interfaz no necesaria, leak de datos, solo se manejan las IDs en frontend

FranWDev commented on frontend/src/app/interfaces/Recipe/Recipe.ts:
interfaz no necesaria, leak de datos, solo se manejan las IDs en frontend

FranWDev commented on frontend/src/app/interfaces/Recipe/RecipeAllergen.ts:
interfaz no necesaria, leak de datos, solo se manejan las IDs en frontend

FranWDev commented on frontend/src/app/interfaces/Request/AllergenRequestDTO.ts:
DTO correcto

FranWDev commented on frontend/src/app/interfaces/Request/OrderRequestDTO.ts:
DTO correcto

FranWDev commented on frontend/src/app/interfaces/Request/RecipeRequestDTO.ts:
DTO Correcto

FranWDev commented on frontend/src/app/interfaces/Response/LoginResponseDTO.ts:
DTO correcto, este te va a dar pelea cuando nos pongamos con la seguridad btw

FranWDev commented on frontend/src/app/interfaces/Response/OrderDetailResponseDTO.ts:
DTO Correcto

FranWDev commented on frontend/src/app/interfaces/Response/OrderResponseDTO.ts:
DTO correcto

FranWDev commented on frontend/src/app/interfaces/Response/ProductResponseDTO.ts:
DTO correcto

FranWDev commented on frontend/src/app/interfaces/Response/RecipeAuditResponseDTO.ts:
DTO correcto

FranWDev reviewed (changes requested):
Eliminar lo que no este relacionado con requests y responses, ya que no cabe posibilidad de problemas N+1 al procesar información de recetas o alergenos en el front, ya se solventaron en el backend, solo hay un pequeño error en:
`frontend/src/app/interfaces/Response/InventoryMovementResponseDTO.ts` 
UserName y UserId estan invertidos.

user-ijavieh committed:
Creación de interfaces de los DTO

Se han borrado los archivos innecesarios y corregido el error en InventoryMovementResponseDTO.ts, invirtiendo los campos de UserId y UserName

user-ijavieh committed:
Archivo Pageable añadido

Se volvió a añadir el archivo de Pageable.ts, previamente eliminado por error

FranWDev reviewed (approved):
Request y Respones correctas, perfecto
